src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -182,6 +237,25 @@ public static void disableCache() {
     return resolveRawArguments(resolveGenericType(type, subType), subType);
   }
 
+  /**
+   * The same as {@link #substituteGenerics(Type, Class)}, but first resolves the generic type of {@code type}.
+   * This is a convenience method.
+   */
+  public static <T, S extends T> Type substituteGenerics(Class<T> type, Class<S> subType) {
+    return substituteGenerics(resolveGenericType(type, subType), getTypeVariableMap(subType, null));
+  }
+
+  /**
+   * Traverses a generic type and replaces all type variables.
+   *
+   * @param type the generic type to traverse
+   * @param context the class that serves as starting point to resolve replacements of type variables
+   * @return a type that is structurally the same as {@code type}, except that type variables have been replaced
+   */
+  public static Type substituteGenerics(Type type, Class<?> context) {
```

Comment:
Same re: `@throws UnsupportedOperationException`

Reply:
Resolved in fc7cffbf0ae21d21b609c6e407458b3e16d5b474